### PR TITLE
[WIP][3.4.x] User activation timeout fix

### PR DESCRIPTION
**Pull request by desoliture1**
## Pull Request description

Issue: #6710 

The cause of the problem is that `timeout` parameter was not used at all, when we use invalid data (host, port, etc.) - mail sender throwing exception immediately, but if the host and port are correct but something else (login or password) is invalid - we just waiting for default timeout (around 2 mins) and timeout from sysadmin configs is not being used

Add *submitWithTimeout* method to `DonAsynchron` class
Add support for timeout in sending messages
Add async sending the 'account activated' email
Add corresponding tests

## General checklist

- [x] You have reviewed the guidelines [document](https://docs.google.com/document/d/1wqcOafLx5hth8SAg4dqV_LV3un3m5WYR8RdTJ4MbbUM/edit?usp=sharing).
- [x] PR name contains fix version. For example, "[3.3.4] Hotfix of some UI component" or "[3.4] New Super Feature".
- [x] Description references specific [issue](https://github.com/thingsboard/thingsboard/issues).
- [x] Description contains human-readable scope of changes.
- [x] Description contains brief notes about what needs to be added to the documentation.
- [x] No merge conflicts, commented blocks of code, code formatting issues.
- [x] Changes are backward compatible or upgrade script is provided.
- [x] Similar PR is opened for PE version to simplify merge. Required for internal contributors only.

## Back-End feature checklist

- [x] Added corresponding unit and/or integration test(s). Provide written explanation in the PR description if you have failed to add tests.
- [x] If new dependency was added: the dependency tree is checked for conflicts.
- [x] If new service was added: the service is marked with corresponding @TbCoreComponent, @TbRuleEngineComponent, @TbTransportComponent, etc.
- [x] If new REST API was added: the RestClient.java was updated, issue for [Python REST client](https://github.com/thingsboard/thingsboard-python-rest-client) is created.

### CODE REVIEW

File: application/src/main/java/org/thingsboard/server/service/mail/DefaultMailService.java [hunk 1]
```diff
@@ -77,10 +86,14 @@ public class DefaultMailService implements MailService {
     @Autowired
     private MailExecutorService mailExecutorService;
 
+    private ScheduledExecutorService timeoutExecutor = Executors.newSingleThreadScheduledExecutor();
```
Comment (ashvayka) [inline application/src/main/java/org/thingsboard/server/service/mail/DefaultMailService.java:89]: This executor is not destroyed in the PreDestroy of the Service

File: common/util/src/main/java/org/thingsboard/common/util/DonAsynchron.java [hunk 2]
```diff
@@ -65,4 +67,32 @@ public static <T> ListenableFuture<T> submit(Callable<T> task, Consumer<T> onSuc
...
     }
 
+    public static <T> ListenableFuture<T> submitWithTimeout(
+            Callable<T> task,
+            long timeout, TimeUnit timeUnit,
+            Executor executor,
+            ScheduledExecutorService timeoutExecutor
+    ) {
+        var future = Futures.submit(task, executor);
+        Futures.withTimeout(future, timeout, timeUnit, timeoutExecutor);
```
Comment (ashvayka) [inline common/util/src/main/java/org/thingsboard/common/util/DonAsynchron.java:77]: Completely wrong. You should return the result of Futures.withTimeout. Also, this method is completely redundant

File: common/util/src/main/java/org/thingsboard/common/util/DonAsynchron.java [hunk 3]
```diff
@@ -65,4 +67,32 @@ public static <T> ListenableFuture<T> submit(Callable<T> task, Consumer<T> onSuc
...
+        return future;
+    }
+
+    public static <T> ListenableFuture<T> submitWithTimeout(
```
Comment (ashvayka) [inline common/util/src/main/java/org/thingsboard/common/util/DonAsynchron.java:81]: remove changes to this file. they are not needed.

File: common/util/src/main/java/org/thingsboard/common/util/DonAsynchron.java (hunk 2 above)
Comment (desoliture1) [reply common/util/src/main/java/org/thingsboard/common/util/DonAsynchron.java:77]: There is no difference what we'll return - Futures.withTimeout will cancel the future-delegate, so we will getting the exception either on 'future' or Futures.withTimeout

**Comment by desoliture1** [comment]
#6862

**Pull request closed without merging**